phpunit/class-rest-block-types-controller-test.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * REST API: REST_Block_Types_Controller_Test class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Tests for REST API for Widgets.
+ *
+ * @see WP_Test_REST_Controller_Testcase
+ */
+class REST_Block_Types_Controller_Test extends WP_Test_REST_TestCase {
+	/**
+	 * @var int
+	 */
+	protected static $admin_id;
+
+	/**
+	 * Create fake data before our tests run.
+	 *
+	 * @param WP_UnitTest_Factory $factory Helper that lets us create fake data.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$admin_id = $factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+		wp_register_script( 'core-assets-test-script', home_url( '/assets/test-script.js' ) );
+		wp_register_script( 'core-assets-test-editor-script', home_url( '/assets/test-editor-script.js' ) );
+		wp_register_style( 'core-assets-test-style', home_url( '/assets/test-style.css' ) );
+		wp_register_style( 'core-assets-test-editor-style', home_url( '/assets/test-editor-style.css' ) );
```

Comment:
One thing I would love to test here is weird inter dependencies. What happens when you have something that has dependencies of a dependencies. How does WordPress resolve that?

Reply:
@spacedmonkey Okay. I will try to re-implement the approach from scripts/styles controllers test where there are nested deps.

Reply:
No, I think you are good with what you have. I would have a new test with really messed on dependencies. Like three levels, just to see what happens and how it breaks.

Reply:
@spacedmonkey Not sure if that test would apply here. Block types tests are only to test for proper `_links` entry to be returned like following example:

```
"wp:editor_script": [
            {
                "embeddable": true,
                "href": "http://localhost:8889/wp-json/wp/v2/scripts/core-assets-test"
            }
        ],
```

We don't seem to be returning here any dependencies or deeper relationships.